packages/snaps-controllers/src/cronjob/CronjobController.ts

```diff
@@ -267,24 +335,131 @@ export class CronjobController extends BaseController<
   }
 
   /**
-   * Unregister all jobs related to the given snapId.
+   * Schedule a background event.
+   *
+   * @param backgroundEventWithoutId - Background event.
+   * @returns An id representing the background event.
+   */
+  scheduleBackgroundEvent(
+    backgroundEventWithoutId: Omit<BackgroundEvent, 'id' | 'scheduledAt'>,
+  ) {
+    const event = {
+      ...backgroundEventWithoutId,
+      id: nanoid(),
+      scheduledAt: new Date().toISOString(),
```

Comment:
This somewhat gives Snaps a high-precision timer, which we are trying to avoid by attenuating `Date`. We should probably truncate as described in the SIP

Reply:
Done.